### PR TITLE
Read _JAVA_OPTIONS env var instead of JAVA_OPTIONS

### PR DESCRIPTION
**Pull request by pshipton**
https://github.com/eclipse-openj9/openj9/pull/10175 added support for
JAVA_OPTIONS for "compatibility" with Hotspot, but this variable is not
used by Hotspot. There is a _JAVA_OPTIONS variable as mentioned in
https://bugs.openjdk.java.net/browse/JDK-8170832 which is recognized by
Java 8+, which is processed after command line arguments.

Fixes https://github.com/eclipse-openj9/openj9/issues/12797

### CODE REVIEW

**Comment by pshipton** [comment]
jenkins compile osx,win jdk11

**Comment by pshipton** [comment]
Tested via grinder, which passed https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/39/

**Comment by pshipton** [comment]
Actually this needs more work, as _JAVA_OPTIONS overrides command line arguments.

**Comment by keithc-ca** [comment]
Note also https://bugs.openjdk.java.net/browse/JDK-8173712 which renamed the variable to `JDK_JAVA_OPTIONS`.

**Comment by pshipton** [comment]
jenkins compile win jdk11

**Comment by pshipton** [comment]
> Note also https://bugs.openjdk.java.net/browse/JDK-8173712 which renamed the variable to JDK_JAVA_OPTIONS

JDK_JAVA_OPTIONS is already supported with OpenJ9, as this option is processed by the launcher itself.

**Comment by pshipton** [comment]
jenkins compile win jdk11

**Comment by pshipton** [comment]
jenkins compile win jdk11

**Comment by pshipton** [comment]
Tested via grinder, which passed https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/45

File: runtime/oti/vmargs_api.h [hunk 1]
```diff
@@ -217,6 +217,17 @@ addJavaPropertiesOptions(J9PortLibrary * portLib, J9JavaVMArgInfoList *vmArgumen
...
+/**
+ * Add the arguments coming from a specific environment variable
+ * @param portLib port library
+ * @param envVarName environment variable name
+ * @param vmArgumentsList current list of arguments
+ * @param verboseFlags set to VERBOSE_INIT for verbosity
+ * @return 0 on success, negative value on failure
+ */
+IDATA
+addEnvironmentVariableArguments(J9PortLibrary* portLib, const char* envVarName, J9JavaVMArgInfoList *vmArgumentsList, UDATA verboseFlags);
```
Comment (keithc-ca) [inline runtime/oti/vmargs_api.h:229]: Please format consistently (`*` binds more tightly to parameter name than the type, like `vmArgumentsList`); use same format in `vmargs.c`.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java [hunk 2]
```diff
@@ -719,14 +719,11 @@ public void testEnvironmentVariableOrdering() {
 				argumentPositions.containsKey(openJ9JavaOptionsArg));
 		assertTrue(IBM_JAVA_OPTIONS+ SHOULD_COME_AFTER+OPENJ9_JAVA_OPTIONS, 
 				argumentPositions.get(ibmJavaOptionsArg).intValue() > argumentPositions.get(openJ9JavaOptionsArg).intValue());
+		assertTrue(OPENJ9_JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_TOOL_OPTIONS, 
+				argumentPositions.get(openJ9JavaOptionsArg).intValue() > argumentPositions.get(javaToolOptionsArg).intValue());
 		if (!isJava8 || !isIBM) {
-			assertTrue(OPENJ9_JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_OPTIONS,
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: This `if` would read better if inverted (`if (isJava8 && isIBM) { ...`).

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: Do you mean `if (!(isJava8 && isIBM)) {` ?

**Comment by pshipton** [comment]
Grinder on latest test changes passed https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/46/

File: runtime/oti/vmargs_api.h (hunk 1 above)
Comment (pshipton) [reply runtime/oti/vmargs_api.h]: Updated.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 2 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: Updated.

Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: No, I meant 'invert' like you did with the `if` statement now on line 348.

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: There is no else case for this one.

Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: Right - I guess I missed that you made a branch of that `if` statement unconditional.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java [hunk 3]
```diff
@@ -719,14 +719,11 @@ public void testEnvironmentVariableOrdering() {
...
-		if (!isJava8 || !isIBM) {
-			assertTrue(OPENJ9_JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_OPTIONS, 
-					argumentPositions.get(openJ9JavaOptionsArg).intValue() > argumentPositions.get(javaOptionsArg).intValue());
-			assertTrue(JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_TOOL_OPTIONS, 
-					argumentPositions.get(javaOptionsArg).intValue() > argumentPositions.get(javaToolOptionsArg).intValue());
-		} else {
-			assertTrue(OPENJ9_JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_TOOL_OPTIONS, 
-					argumentPositions.get(openJ9JavaOptionsArg).intValue() > argumentPositions.get(javaToolOptionsArg).intValue());
+		assertTrue(OPENJ9_JAVA_OPTIONS + SHOULD_COME_AFTER + JAVA_TOOL_OPTIONS, 
+				argumentPositions.get(openJ9JavaOptionsArg).intValue() > argumentPositions.get(javaToolOptionsArg).intValue());
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: Is this change intentional? It only matches an old test for `isJava8 && isIBM`.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java [hunk 4]
```diff
@@ -719,14 +719,11 @@ public void testEnvironmentVariableOrdering() {
...
+		if (!(isJava8 && isIBM)) {
+			assertTrue(JAVA_OPTIONS + SHOULD_COME_AFTER + IBM_JAVA_OPTIONS, 
+					argumentPositions.get(javaOptionsArg).intValue() > argumentPositions.get(ibmJavaOptionsArg).intValue());
```
Comment (keithc-ca) [inline test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: This doesn't seem right. Why would we expect a `!isIBM` VM to pay attention to `IBM_JAVA_OPTIONS`?

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 3 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: Since JAVA_OPTIONS is removed and _JAVA_OPTIONS is parsed after command line options, OpenJ9 and IBM Java 8 now have the same behaviour, OPENJ9_JAVA_OPTIONS comes after JAVA_TOOL_OPTIONS.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 4 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: IBM_JAVA_OPTIONS is and has always been parsed on OpenJ9. We added OPENJ9_JAVA_OPTIONS as the replacement, deprecating IBM_JAVA_OPTIONS, but it still works.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 3 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: OPENJ9_JAVA_OPTIONS has always been after JAVA_TOOL_OPTIONS, but previously it was also after JAVA_OPTIONS (on OpenJ9 builds) which is why the test was conditional.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 4 above)
Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: The only difference between IBM Java 8 and other builds related to these changes is that IBM Java 8 doesn't parse _JAVA_OPTIONS and OpenJ9 builds do.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 3 above)
Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:723]: A build that identified as `!isIBM` should not be paying any attention to `OPENJ9_JAVA_OPTIONS`.

File: test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java (hunk 4 above)
Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: I was looking at [`redirector.c`](https://github.com/eclipse-openj9/openj9/blob/master/runtime/redirector/redirector.c#L565) where it ignores `IBM_JAVA_OPTIONS` if `OPENJ9_JAVA_OPTIONS` is defined. Do you think that's inconsisten?

Comment (pshipton) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: It is weird that we check options in the redirector that way, and then process both environment variables in the VM. I'm not too concerned about it unless somebody complains. If we manage to switch IBM to use mixed builds then the redirector code becomes obsolete.

Comment (keithc-ca) [reply test/functional/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java:726]: The redirector is looking for options like `-Xnocompressedrefs`.

**Review by keithc-ca — APPROVED** [review]
